src/java/org/apache/cassandra/db/memtable/SkipListMemtable.java

```
@@ -0,0 +1,391 @@
+{
+    private static final Logger logger = LoggerFactory.getLogger(SkipListMemtable.class);
+
+    public static final Factory FACTORY = SkipListMemtableFactory.INSTANCE;
+
+    private static final int ROW_OVERHEAD_HEAP_SIZE;
+    static
+    {
+        int userDefinedOverhead = MEMTABLE_OVERHEAD_SIZE.getInt(-1);
+        if (userDefinedOverhead > 0)
+            ROW_OVERHEAD_HEAP_SIZE = userDefinedOverhead;
+        else
+            ROW_OVERHEAD_HEAP_SIZE = estimateRowOverhead(MEMTABLE_OVERHEAD_COMPUTE_STEPS.getInt());
+    }
+
+    // We index the memtable by PartitionPosition only for the purpose of being able
+    // to select key range using Token.KeyBound. However put() ensures that we
+    // actually only store DecoratedKey.
+    private final ConcurrentNavigableMap<PartitionPosition, AtomicBTreePartition> partitions = new ConcurrentSkipListMap<>();
+
+    private final AtomicLong liveDataSize = new AtomicLong(0);
+
+    SkipListMemtable(AtomicReference<CommitLogPosition> commitLogLowerBound, TableMetadataRef metadataRef, Owner owner)
+    {
+        super(commitLogLowerBound, metadataRef, owner);
+    }
+
+    // Only for testing
+    @VisibleForTesting
+    public SkipListMemtable(TableMetadataRef metadataRef)
+    {
+        this(null, metadataRef, new Owner()
+        {
+            @Override
+            public ListenableFuture<CommitLogPosition> signalFlushRequired(Memtable memtable, ColumnFamilyStore.FlushReason reason)
+            {
+                return null;
+            }
+
+            @Override
+            public Memtable getCurrentMemtable()
+            {
+                return null;
+            }
+
+            @Override
+            public Iterable<Memtable> getIndexMemtables()
+            {
+                return Collections.emptyList();
+            }
+
+            public ShardBoundaries localRangeSplits(int shardCount)
+            {
+                return null; // not implemented
```

Maybe we should throw `UnsupportedOperationException`

---

The only callsite actually had access to the CFS, this is not necessary any more.